### PR TITLE
Remove trait for large value persistence

### PR DESCRIPTION
**Pull request by siwane**
The purpose of this PR is to refactor the large KV persistence. It is now merged to KV and AdvKV default persistences. 

> "large KV persistence" means that a persistence can handle value bigger than it can originally handles. For example dynamo is limited to a value size of 400KB, with this feature, value size is unlimited

Ideally use alternative result storage (https://hub.taocloud.org/articles/administrator-guide/data-abstractions > " Result storage abstraction")
After pull, change your persistence parameter and add : 
```
        'keyValueResult' => array(
            'driver' => 'phpredis',
            'host' => '127.0.0.1',
            'port' => 6379,
            'max_value_size' => 500
        )
```

### CODE REVIEW

File: test/common/persistence/KeyValuePersistenceTest.php [hunk 1]
```diff
@@ -19,20 +19,32 @@
 
 namespace oat\generis\test\common\persistence;
 
-class KeyLargeValuePersistenceTest extends \PHPUnit_Framework_TestCase
+use \PHPUnit_Framework_TestCase as TestCase;
+
+class KeyValuePersistenceTest extends TestCase
```
Comment (bugalot) [inline test/common/persistence/KeyValuePersistenceTest.php]: Launching this test case with Redis driver provides me the following errors in PHPUnit:

```
1) oat\generis\test\common\persistence\KeyValuePersistenceTest::testDelExistsLarge
Failed asserting that true is false.

/home/jerome/Development/TAO/taoplatform300/generis/test/common/persistence/KeyValuePersistenceTest.php:78

2) oat\generis\test\common\persistence\KeyValuePersistenceTest::testDelExists
Failed asserting that true is false.

/home/jerome/Development/TAO/taoplatform300/generis/test/common/persistence/KeyValuePersistenceTest.php:89

3) oat\generis\test\common\persistence\KeyValuePersistenceTest::testMapMapControl
Failed asserting that true is false.

/home/jerome/Development/TAO/taoplatform300/generis/test/common/persistence/KeyValuePersistenceTest.php:78
/home/jerome/Development/TAO/taoplatform300/generis/test/common/persistence/KeyValuePersistenceTest.php:110
```

Comment (bugalot) [reply test/common/persistence/KeyValuePersistenceTest.php]: I just commited something to make it easier to test with other drivers.

Comment (siwane) [reply test/common/persistence/KeyValuePersistenceTest.php]: Solved, inMemoryDriver is not persistent, Redis is. It's why in the second test, the driver already have the key 'test'

File: common/persistence/class.KeyValuePersistence.php [hunk 2]
```diff
@@ -14,46 +14,374 @@
...
+     * @param $value
+     * @param null $ttl
+     * @return bool
+     * @throws common_Exception If size is misconfigured
+     */
     public function set($key, $value, $ttl = null)
     {
+        if ($this->isLarge($value)) {
+            common_Logger::t('Large value detected into KeyValue persistence. Splitting value for key : ' . $key);
+            $value = $this->setLargeValue($key, $value);
```
Comment (bugalot) [inline common/persistence/class.KeyValuePersistence.php]: What about TTL here ?

File: common/persistence/class.KeyValuePersistence.php [hunk 3]
```diff
@@ -14,54 +14,421 @@
...
+    {
+        $value = $this->getDriver()->get($key);
+        if ($this->isSplit($value)) {
+            common_Logger::t('Large value detected into KeyValue persistence. Joining value for key : ' . $key);
+            $value = $this->join($key, $value);
+        }
+        return $value;
     }
 
-    public function incr($key) {
```
Comment (bugalot) [inline common/persistence/class.KeyValuePersistence.php]: `incr` and `decr` are now missing from class KeyValuePersistence which is causing trouble.

**Pull request merged**